### PR TITLE
Book detail

### PR DESCRIPTION
**Pull request by lfmontesanto**
**Summary:** 

- Stack Navigation from library to book detail

- Book detail screen with review list

- Colors file

- Fonts file

**ScreenShot** 

Screenshot_1581719801

Screenshot_1581719756

### CODE REVIEW

File: Wbooks/src/app/screens/bookDetail/components/bookInformation/index.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            uri: imageUrl
+          }}
+          style={styles.bookCover}
+        />
+        <View style={styles.bookDetailContainer}>
+          <Text style={styles.title} ellipsizeMode="tail">
+            {title}
+          </Text>
+          <Text style={available ? styles.bookAvailable : styles.bookAvailable}>
+            {available ? 'Available' : 'Not Available'}
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/bookDetail/components/bookInformation/index.js]: {`${!available && 'Not'}` && 'Available'} try this

File: Wbooks/src/app/screens/bookDetail/components/bookInformation/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import { View, Text, Image } from 'react-native';
+import PropTypes from 'prop-types';
+
+import BookDetailButton from '../button/index';
+import bookModel from '../../../../proptypes/bookModel';
+
+import styles from './styles';
+
+function BookInformation({ book, available }) {
+  const { image_url: imageUrl, title, author, genre, year } = book;
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/bookDetail/components/bookInformation/index.js]: Destruct this on function declaration.

File: Wbooks/src/app/screens/bookDetail/components/bookInformation/styles.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { StyleSheet } from 'react-native';
+
+import colors from '../../../../../constants/colors';
+import fonts from '../../../../../constants/fonts';
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/bookDetail/components/bookInformation/styles.js]: Import just what you need.

File: Wbooks/src/app/screens/bookDetail/components/bookInformation/styles.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+const styles = StyleSheet.create({
+  bookInfoContainer: {
+    marginHorizontal: 20,
+    marginTop: 20,
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/bookDetail/components/bookInformation/styles.js]: Declare repeated values as constants and replace.
```
const MEDIUM_MARGIN = 20
const LARGE_MARGIN = 30
...
```

Comment (rodrigoWolox) [reply Wbooks/src/app/screens/bookDetail/components/bookInformation/styles.js]: Do the same with the rest key-values

File: Wbooks/src/app/screens/bookDetail/components/button/index.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import { View, Text } from 'react-native';
+import PropTypes from 'prop-types';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+
+import styles from './styles';
+
+function BookDetailButton({ solid, title }) {
+  return (
+    <View style={solid ? styles.solidButton : styles.shallowButton}>
+      <TouchableOpacity>
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/bookDetail/components/button/index.js]: Is this component necessary?

File: Wbooks/src/app/screens/bookDetail/components/button/styles.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import { StyleSheet } from 'react-native';
+
+import colors from '../../../../../constants/colors';
+import fonts from '../../../../../constants/fonts';
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/bookDetail/components/button/styles.js]: Import just what you need

File: Wbooks/src/app/screens/bookDetail/components/button/styles.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    textAlign: 'center'
+  },
+  solidButtonText: {
+    fontFamily: fonts.sfHeavy,
+    color: colors.white,
+    fontSize: 15,
+    fontWeight: 'bold',
+    lineHeight: 19,
+    textAlign: 'center'
+  }
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/bookDetail/components/button/styles.js]: Declare constants for repeated values and replace.

File: Wbooks/src/app/screens/bookDetail/components/review/styles.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
+import { StyleSheet } from 'react-native';
+
+import colors from '../../../../../constants/colors';
+import fonts from '../../../../../constants/fonts';
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/bookDetail/components/review/styles.js]: Same as before and don't forget the constants for repeated values.

File: Wbooks/src/app/screens/bookDetail/components/reviewsList/index.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+import reviewModel from '../../../../proptypes/reviewModel';
+import Review from '../review';
+
+import styles from './styles';
+
+class ReviewsList extends Component {
+  renderItem = ({ item }) => <Review review={item} />;
+
+  keyExtractor = item => item.id.toString();
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/bookDetail/components/reviewsList/index.js]: Use a template instead `.toString()` `keyExtractor = item => ${item.id};`

File: Wbooks/src/app/screens/bookDetail/index.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+import reviewModel from '../../proptypes/reviewModel';
+
+import BookInformation from './components/bookInformation';
+import styles from './styles';
+import ReviewsList from './components/reviewsList';
+import reviewsMock from './reviewsmock.json';
+
+function BookDetailContainer({ route }) {
+  const { book } = route.params;
+  const reviews = reviewsMock;
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/bookDetail/index.js]: I think this is not necessary, you can pass `reviewsMock` as the value for `reviews` prop on line 19

File: Wbooks/src/app/screens/library/components/book/styles.js [hunk 11]
```diff
@@ -1,7 +1,8 @@
 import { StyleSheet } from 'react-native';
 
+import fonts from '../../../../../constants/fonts';
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/book/styles.js]: Import just what you need and replace

File: Wbooks/src/app/screens/library/components/bookList/index.js [hunk 12]
```diff
@@ -1,19 +1,18 @@
...
 
 class BookListContainer extends Component {
-  renderItem = ({ item }) => <Book book={item} />;
+  renderItem = ({ item }) => <Book book={item} navigation={this.props.navigation} />;
 
   keyExtractor = item => item.id.toString();
 
   render() {
-    const { bookList } = this.props;
+    const bookList = mockData;
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/bookList/index.js]: Same as before, just pass `mockData` as the prop value

File: Wbooks/src/app/screens/navigator.js [hunk 13]
```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/navigator.js]: For my grandma's underwear!!! You are importing all react here, i think is not necessary. Delete this and try the app, it should work.

Comment (lfmontesanto) [reply Wbooks/src/app/screens/navigator.js]: mlb-reaction-pablo-sandoval-1ipgzFqnollIuA4OdL

File: Wbooks/src/app/screens/library/components/book/index.js [hunk 14]
```diff
@@ -1,24 +1,40 @@
...
-      <View style={styles.descriptionContainer}>
-        <Text style={styles.title}>{title}</Text>
-        <Text style={styles.author}>{author}</Text>
-      </View>
-    </View>
-  );
+class Book extends Component {
+  handlePress = () => {
+    const { navigation, book } = this.props;
+    navigation.navigate('BookDetail', { navigation, book });
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/book/index.js]: Replace `BookDetail` with routes constant

**Pull request merged**